game/scripts/vscripts/util/other.lua

```diff
@@ -180,22 +190,28 @@ function CastAdditionalAbility(caster, ability, target)
 		end
 	end
 	skill:OnSpellStart()
-	if channelled then
-		Timers:CreateTimer(0.03, function()
-			if not caster:IsChanneling() then
-				skill:EndChannel(true)
-				skill:OnChannelFinish(true)
-				Timers:NextTick(function()
-					if skill then UTIL_Remove(skill) end
-					if unit then UTIL_Remove(unit) end
-				end)
-			else
-				return 0.03
+	if channelTime > 0 then
+		local abilityLastEndTime = ability.lastEndTime
+		if ability:IsChanneling() and (not abilityLastEndTime or abilityLastEndTime < GameRules:GetGameTime() - delay) then
+			local index = #ability.EndChannelListeners + 1
+			ability.EndChannelListeners[index] = function(interrupted)
+				ability.EndChannelListeners[index] = nil
+				EndAdditionalAbilityChannel(caster, unit, skill, interrupted, delay)
 			end
-		end)
+		else
+			EndAdditionalAbilityChannel(caster, unit, skill, ability.channelFailed, delay - GameRules:GetGameTime() + abilityLastEndTime)
```

Comment:
What is the situation where this would happen?

Reply:
when you refresher powershot before you finish multicasting powershot